File: backend/app/schemas/task.py
from datetime import datetime

from pydantic import BaseModel


class Task(BaseModel):
    scheduled_task_id: int
    maintenance_task_id: int

    schedule_code: str

    device: str
    task_code: str

    description: str | None = None

    frequency_id: int | None = None
    frequency: str | None = None
    frequency_num_of_days: int | None = None

    planned_date: datetime

    tolerance_start: datetime | None = None
    tolerance_end: datetime | None = None
    tolerance_status: str | None = None

    status: str
    status_id: int

    remarks: str | None = None
    done_at: datetime | None = None
    performed_by: int | None = None
    performance_remarks: str | None = None

File: frontend/src/types/task.ts
export interface Task {
  scheduled_task_id: number;
  maintenance_task_id: number;

  schedule_code: string;

  device: string;
  task_code: string;

  description: string | null;

  frequency_id: number | null;
  frequency: string | null;
  frequency_num_of_days: number | null;

  planned_date: string;

  tolerance_start: string | null;
  tolerance_end: string | null;
  tolerance_status:
    | "out_of_tolerance"
    | "in_tolerance"
    | "upcoming"
    | "unknown_tolerance"
    | null;

  status: string;
  status_id: number;

  remarks: string | null;
  done_at: string | null;
  performed_by: number | null;
  performance_remarks: string | null;
}